ml_model/src/main/java/edu/wm/sealab/featureextraction/Parser.java

```diff
@@ -105,11 +145,33 @@ public static void main(String[] args) {
                 pw.append(Integer.toString(features.getParenthesis()));
                 pw.append(",");
                 pw.append(Integer.toString(features.getLiterals()));
+                pw.append(",");
+                pw.append(Double.toString(avgNumOfComments));
+                pw.append(",");
+                pw.append(Double.toString(avgNumOfComparisons));
+                pw.append(",");
+                pw.append(Double.toString(avgNumOfArithmeticOperators));
+                pw.append(",");
+                pw.append(Double.toString(avgNumOfConditionals));
                 pw.append("\n");
               })
           .explore(projectDir);
     } catch (FileNotFoundException e) {
       e.printStackTrace();
     }
   }
+
+  private static int findCorrespondingEntry(List<String[]> lines, String fileName) {
```

Comment:
This method's purpose is not obvious. Please write a short Javadoc comment that explains what it does, and why you can't use a method from the standard library to do this instead.

Reply:
It simply searches the csv file for the correct entry that matches the file name of whatever file the DirExplorer is currently parsing. If there is a better way to do this, I would be happy to learn it!

Reply:
If you prefer, I can move that code to inside the DirExplorer. I just thought it made Parser.java neater and easier to read to make a separate private method.

Reply:
Thanks! I wasn't sure if this was implementing a standard search of some kind that we could use the standard library for, or whether it was some custom algorithm. It looks like this probably the latter (or at least easiest to model as the latter), and your new comment explains it well.